jenkins/image_building/initrd_sdk/unlock-mount-luks.sh

```diff
@@ -0,0 +1,142 @@
+    done < "/proc/partitions"
+    printf "%s" "${_count}"
+}
+
+# Start of the execution
+
+# The script has 1 mandatory and 3 optional positional arguments
+# Such as:
+# - path to the device file of the root partition (mandatory)
+# - path to the script that provides the encryption key in plain text format
+# - the partition number of the config-drive
+# - flag to run in dry run mode (nothig will get created/unlocked/mounted
+root_device_path="${1:?}"
+key_script="${2:-/etc/tpm2-unseal-key.sh}"
+config_drive_part_num="${3:-}"
+dry_run="${4:-false}"
+
+printf "INFO: unlock script has been started with the following arguments:\n"
+printf "Root device:%s, key script:%s, config-drive partition number:%s, " \
+    "${root_device_path}" "${key_script}" "${config_drive_part_num}"
+printf "dry run:%s\n" "${dry_run}"
+
+# create the mount point for the root file system
+if [[ "${dry_run}" == "false" ]]; then
+    mkdir "/realroot"
+fi
+
+# different workflows depending on the presence of encryption
+# --------------------------------------------------------------------
+# nvme, mmc/sd card and loop devices have a partition number prefix
+# e.g./dev/nvme0n1 <--> /dev/nvme0n1p1
+# --------------------------------------------------------------------
+# It is expected that the last partition on the root device is the
+# config drive partition thus that is the default logic. In case the user
+# specified a value for `config_drive_part_num` the partition count will
+# be discarded and the value of `config_drive_part_num` will be used instead.
+if [[ $(_is_luks "${root_device_path}") ]]; then
+        printf "Mounting encrypted %s\n" "${root_device_path}"
+        if [[ "${dry_run}" == "false" ]]; then
+            /usr/lib/systemd/systemd-cryptsetup attach realroot \
+                "${root_device_path}" <("${key_script}") luks
+            mount "/dev/mapper/realroot" "/realroot"
+        fi
+        root_partition="$(_get_partition_from_blkid "${root_device_path}")"
+        part_prefix="$(_get_part_prefix_for_device "${root_partition}")"
+        root_device="${root_partition%"${part_prefix}"*}"
+        part_count="$(_count_parts_for_disk "${root_device}")"
+        config_drive_common="/dev/${root_device}${part_prefix}"
+        if [[ -z "${config_drive_part_num}" ]]; then
+            config_drive_path="${config_drive_common}${part_count}"
+        else
+            config_drive_path="${config_drive_common}${config_drive_part_num}"
+        fi
+        if [[ $(_is_luks "${config_drive_path}") ]]; then
+            printf "Unlocking config drive %s\n" "${config_drive_path}"
+            printf "INFO: config-drive:%s disk:%s part_count:%s " \
+                "${config_drive_path}" "${root_device}" "${part_count}"
+            printf "root_partition:%s prefix:%s\n" "${root_partition}" \
+                "${part_prefix}"
+            if [[ "${dry_run}" == "false" ]]; then
+                /usr/lib/systemd/systemd-cryptsetup attach "config-2" \
```

Comment:
With that, this would become 

```suggestion
                _dry_run /usr/lib/systemd/systemd-cryptsetup attach "config-2" \
```